### ISSUE TITLE
Mos 2.19.0 : Filename conflict on .o files during build

### ISSUE THREAD

**Issue by Harvie** [mongoose-os/mos#50]
Hello,
i use 3rd party library which contains file `./deps/mylib/src/mylib_hal/esp/platform.c`
mongoose contains mbedtls with file `./deps/modules/mbedtls/library/platform.c`
After upgrade to mos 2.19.0 both of these files compile to .o file `./build/objs/mosapp/platform.c.o`
Later one to be compiled overwrites the previous, this causes linker error.

Can you please prefix the .o files with full path to prevent this? It is really annoying. Eg.:

`./build/objs/mosapp/deps/mylib/src/mylib_hal/esp/platform.c.o`
`./build/objs/mosapp/deps/modules/mbedtls/library/platform.c.o`
or
`./build/objs/mosapp/deps_mylib_src_mylib_hal_esp_platform.c.o`
`./build/objs/mosapp/deps_modules_mbedtls_library_platform.c.o`

This would account even for posibility that single library has multiple `platform.c` files in several different directories. (this actually happens in that library as well)
I would rename the files, but i want to include them from upstream git using `modules:` in `mos.yml`.

**Comment by rojer** [comment]
yes, it's a known issue... should be fixed, for sure.

**Comment by rojer** [comment]
pretty nasty job to rewrite the makefiles though... as a short-term solution, your best bet is to rename the files in question

**Comment by Harvie** [comment]
> pretty nasty job to rewrite the makefiles though... as a short-term solution, your best bet is to rename the files in question

yes. i've done that. but i have plans to load 3rd party libraries directly from their upstream git repository which is not possible with this bug.
